### ISSUE TITLE
`find_homography_lines_dlt` pairs point i with point N+i, not a segment's endpoints; slid endpoints give H 3.5e3 px off

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4866]
`find_homography_lines_dlt(ls1, ls2)` takes segments of shape `(B, N, 2, 2)` = `[start, end] × (x, y)`, flattens them to `(B, 2N, 2)` with `reshape` and splits that list in halves with `torch.chunk(dim=1)`. Row pair i of the DLT is therefore built from flattened points i and N + i, not from the two endpoints of segment i: with N = 6, "segment 0" is the line from the start of segment 0 to the start of segment 3. The image-2 lines are built the same way, so the estimate is right only when the endpoints are themselves point correspondences, because any two matched points define a matched line. For a line detector's output, where each image-2 segment lies on the true line but its endpoints sit elsewhere along it, the estimated `H` is 3.5e3 px from the true homography, and `RANSAC("homography_from_linesegments")` returns no model. The formulation the function cites (`homolines2001` and the unizar.es paper linked in the source) needs no endpoint correspondence: both endpoints of segment i in image 1 must map onto the line of segment i in image 2, `l2_iᵀ H x = 0`.

## Repro

```python
import torch
from kornia.geometry.homography import find_homography_lines_dlt
from kornia.geometry.linalg import transform_points
from kornia.geometry.ransac import RANSAC

H_true = torch.tensor([[[1.034685, -0.04160598, -63.61482],
                        [0.1039753, 1.135451, -62.87412],
                        [2.851519e-04, 1.404178e-04, 1.0]]])
# six segments in image 1: (N, 2 endpoints, xy)
ls1 = torch.tensor([[[318.8, 279.9], [194.8, 137.8]],
                    [[316.9, 317.0], [362.2, 154.2]],
                    [[153.2, 90.1], [325.8, 259.0]],
                    [[204.8, 129.7], [447.6, 170.0]],
                    [[294.1, 216.2], [158.5, 95.9]],
                    [[258.3, 329.9], [242.0, 87.6]]])[None]
# the same six lines in image 2
m0 = transform_points(H_true, ls1[:, :, 0])
m1 = transform_points(H_true, ls1[:, :, 1])
# (a) endpoints are point correspondences: image-2 endpoints = H(image-1 endpoints)
ls2_pts = torch.stack([m0, m1], dim=2)
# (b) a line detector's output: the same image-2 lines, endpoints at other positions along them
a = torch.tensor([0.1, -0.2, 0.3, 0.0, -0.1, 0.2]).view(1, 6, 1)
b = torch.tensor([1.3, 0.9, 1.1, 0.7, 1.2, 0.8]).view(1, 6, 1)
ls2_slid = torch.stack([m0 + a * (m1 - m0), m0 + b * (m1 - m0)], dim=2)

def line_err(H, ls2):  # distance (px) from H(image-1 endpoints) to the image-2 line through ls2
    p = transform_points(H, ls1.reshape(1, 12, 2)).reshape(1, 6, 2, 2)
    s, e = ls2[:, :, 0:1], ls2[:, :, 1:2]
    d = e - s
    n = torch.stack([-d[..., 1], d[..., 0]], -1) / d.norm(dim=-1, keepdim=True)
    return ((p - s) * n).sum(-1).abs().max().item()

pts1 = ls1.reshape(1, 12, 2)
print(f"H_true on the slid segments: max endpoint-to-line distance = {line_err(H_true, ls2_slid):.2g} px")
for name, ls2 in [("endpoint correspondences", ls2_pts), ("endpoints slid along the lines", ls2_slid)]:
    H = find_homography_lines_dlt(ls1, ls2)
    xfer = (transform_points(H, pts1) - transform_points(H_true, pts1)).norm(dim=-1).max().item()
    print(f"{name}: max |H(p) - H_true(p)| over the 12 endpoints = {xfer:.4g} px,"
          f" max endpoint-to-line distance = {line_err(H, ls2):.4g} px")

# what the DLT pairs: ls1.reshape(B, 2N, 2) then chunk(2, dim=1) -> 'segment' i = flattened points i and N + i
start, end = torch.chunk(ls1.reshape(1, 12, 2), 2, dim=1)
r = lambda t: [round(v, 1) for v in t.tolist()]
print("segment 0 is built from", r(start[0, 0]), "and", r(end[0, 0]), "| its endpoints are", r(ls1[0, 0, 0]), "and", r(ls1[0, 0, 1]))

ransac = RANSAC("homography_from_linesegments", inl_th=2.0, seed=0)
print("RANSAC.verify accepts H_true on the slid segments:", ransac.verify(ls1[0], ls2_slid[0], H_true, 2.0**2)[1].tolist())
H_r, inl = ransac(ls1[0], ls2_slid[0])
print("RANSAC on slid segments: model all zeros =", bool((H_r == 0).all()), "| inliers", int(inl.sum()), "of 6")
H_r, inl = ransac(ls1[0], ls2_pts[0])
print("RANSAC on endpoint correspondences: model all zeros =", bool((H_r == 0).all()), "| inliers", int(inl.sum()), "of 6")
```

```text
H_true on the slid segments: max endpoint-to-line distance = 1.1e-05 px
endpoint correspondences: max |H(p) - H_true(p)| over the 12 endpoints = 0.0002222 px, max endpoint-to-line distance = 0.0001186 px
endpoints slid along the lines: max |H(p) - H_true(p)| over the 12 endpoints = 3530 px, max endpoint-to-line distance = 844.6 px
segment 0 is built from [318.8, 279.9] and [204.8, 129.7] | its endpoints are [318.8, 279.9] and [194.8, 137.8]
RANSAC.verify accepts H_true on the slid segments: [True, True, True, True, True, True]
RANSAC on slid segments: model all zeros = True | inliers 0 of 6
RANSAC on endpoint correspondences: model all zeros = False | inliers 6 of 6
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same output), CPU float32, macOS arm64.

The image-2 segments in case (b) lie on the true lines (`H_true` leaves every endpoint at most 1.1e-05 px off them); only the endpoint positions along the lines differ from case (a). `RANSAC.verify` accepts `H_true` on the slid data with all six segments as inliers, so the empty RANSAC result comes from the minimal solver, not from the threshold. The code dates from #1851 (2022).

## Classification

Silent wrong result: no error or warning, and the returned matrix is a valid homography, just not the right one.

Idiom 7 (an axis-order or field-order slip hidden by symmetric test data). Every test fixture (`TestFindHomographyFromLinesDLT` in `tests/geometry/test_homography.py`, `TestRANSACHomographyLineSegments` in `tests/geometry/test_ransac.py`) builds `ls2` as `H` applied to the endpoints of `ls1`. On that data any pairing of points gives matching lines, so the tests cannot detect the wrong pairing.

## Disposition

Focused fix, welcome as a PR: normalise the flattened points as now, then view them back as `(B, N, 2, 2)` and take `start = [:, :, 0]`, `end = [:, :, 1]`, so the image-2 line of segment i constrains both image-1 endpoints of segment i (and `weights[:, i]` weights segment i). Exact endpoint-correspondence data gives the same `H` as today.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Related: #4867 (the line-segment error that RANSAC scores these models with).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
